mslice/models/slice/mantid_slice_algorithm.py

```diff
@@ -82,6 +82,15 @@ def compute_symmetrised(self, scattering_data, boltzmann_dist, y_axis):
         negatives = scattering_data[len(energy_transfer):] * boltzmann_dist[len(energy_transfer):,None]
         return np.concatenate((scattering_data[:len(energy_transfer)], negatives))
 
+    def compute_gdos(self, scattering_data, boltzmann_dist, x_axis, y_axis):
+        energy_transfer = np.linspace(y_axis.end, y_axis.start, self._get_number_of_steps(y_axis))
+        momentum_transfer = np.linspace(x_axis.start, x_axis.end, self._get_number_of_steps(x_axis))
+        momentum_transfer = np.square(momentum_transfer[:scattering_data.shape[0]])
```

Comment:
I think we could benefit from using in-place operations here, for example:

```
momentum_transfer = np.square(momentum_transfer, out=momentum_transfer) 
```
will not create a new array but do the operation in place. We can check this with

```
In [20]: x = np.linspace(-100,100, 100)
In [21]: x.__array_interface__['data'][0]
Out[21]: 39198896
In [22]: x = np.square(x, out=x)
In [23]: x.__array_interface__['data'][0]
Out[23]: 39198896
In [24]: x = np.square(x)
In [25]: x.__array_interface__['data'][0]
Out[25]: 39327216
```
where `__array_interface__['data'][0]` is the pointer to the C array. See https://docs.scipy.org/doc/numpy/reference/generated/numpy.square.html